lib/gssapi/krb5/arcfour.c

```diff
@@ -1375,17 +1375,17 @@ _gssapi_unwrap_iov_arcfour(OM_uint32 *minor_status,
 	}
     }
 
+    if (pconf_state) {
+	*pconf_state = conf_state;
+    }
+
     HEIMDAL_MUTEX_lock(&ctx->ctx_id_mutex);
     ret = _gssapi_msg_order_check(ctx->order, seq_number);
     HEIMDAL_MUTEX_unlock(&ctx->ctx_id_mutex);
     if (ret != 0) {
```

Comment:
I think minor_status needs to be set to 0 before this return.  Otherwise there might be random garbage used by the caller.   We might consider setting *minor_status = 0 upon entry to this function instead of setting it at each error return site.